packages/block-library/src/query-pagination-previous/index.php

```diff
@@ -44,13 +45,25 @@ function render_block_core_query_pagination_previous( $attributes, $content, $bl
 		add_filter( 'previous_posts_link_attributes', $filter_link_attributes );
 		$content = get_previous_posts_link( $label );
 		remove_filter( 'previous_posts_link_attributes', $filter_link_attributes );
-	} elseif ( 1 !== $page ) {
-		$content = sprintf(
-			'<a href="%1$s" %2$s>%3$s</a>',
-			esc_url( add_query_arg( $page_key, $page - 1 ) ),
-			$wrapper_attributes,
-			$label
-		);
+	} elseif ( 1 < $page ) {
```

Comment:
Hi @ntsekouras, I was testing out the PR and found another small bug because of the condition `1 !== $page`. Basically, the value of `$page` should always be greater than 1. Otherwise, on negative page numbers (edge case scenario) the previous button is still visible and the navigation is possible infinitely because of the condition. 

Here's a screencast of the same:

https://github.com/user-attachments/assets/5e26432d-747d-4d47-bb42-04fe323b8e72

I went ahead and patched this bug. Although, would love to hear your thoughts on this.

Thanks for reviewing the PR.

Reply:
Oh, yes. It makes sense. Let's keep it!